### ISSUE TITLE
Not able to receive/upload WhatsApp media files

### ISSUE THREAD

**Issue by Lartsch** [StevenBlack/hosts#955]
On Android, users of the **_SOCIAL MEDIA extension_** cannot send or receive any WhatsApp media files, such as documents, voice notes and images. Text messages still work.

I struggled a long time just to find out that it was the hosts file which blocked it. Switching to a hosts file **_without_** the Social Media extension resolves this issue.

I hope someone can confirm this / reproduce it.

**EDIT: Excuse me if this behavior is meant to be like this. I just wondered. Please close the issue if that's the case.**

**Comment by StevenBlack** [comment]
Thanks Lars @Lartsch.  This is an interesting question.  WhatsApp is just enormous.

Several of our curated sources are involved with this.

I need input from other collaborators here...  What does everybody think?

```
data/add.Risk/hosts
505:0.0.0.0 downloads-whatsapp.com
1882:0.0.0.0 www.downloads-whatsapp.com

data/malwaredomainlist.com/hosts
193:127.0.0.1  downloads-whatsapp.com

extensions/social/hosts
1773:# Whatsapp
1774:127.0.0.1 account.whatsapp.com
1775:127.0.0.1 alpha.whatsapp.com
1776:127.0.0.1 api.whatsapp.com
1777:127.0.0.1 autodiscover.whatsapp.com
1778:127.0.0.1 beta.whatsapp.com
1779:127.0.0.1 betablog.whatsapp.com
1780:127.0.0.1 blog.whatsapp.com
1781:127.0.0.1 chat.whatsapp.com
1782:127.0.0.1 crashlogs.whatsapp.net
1783:127.0.0.1 dev-web.whatsapp.com
1784:127.0.0.1 dyn.web.whatsapp.com
1785:127.0.0.1 e1.whatsapp.net
1786:127.0.0.1 e10.whatsapp.net
1787:127.0.0.1 e11.whatsapp.net
1788:127.0.0.1 e12.whatsapp.net
1789:127.0.0.1 e13.whatsapp.net
1790:127.0.0.1 e14.whatsapp.net
1791:127.0.0.1 e15.whatsapp.net
1792:127.0.0.1 e16.whatsapp.net
1793:127.0.0.1 e2.whatsapp.net
1794:127.0.0.1 e3.whatsapp.net
1795:127.0.0.1 e4.whatsapp.net
1796:127.0.0.1 e5.whatsapp.net
1797:127.0.0.1 e6.whatsapp.net
1798:127.0.0.1 e7.whatsapp.net
1799:127.0.0.1 e8.whatsapp.net
1800:127.0.0.1 e9.whatsapp.net
1801:127.0.0.1 faq.whatsapp.com
1802:127.0.0.1 g.whatsapp.net
1803:127.0.0.1 gamma.whatsapp.com
1804:127.0.0.1 graph.whatsapp.net
1805:127.0.0.1 m.whatsapp.com
1806:127.0.0.1 media-ams3-1.cdn.whatsapp.net
1807:127.0.0.1 media-amt2-1.cdn.whatsapp.net
1808:127.0.0.1 media-arn2-1.cdn.whatsapp.net
1809:127.0.0.1 media-ber1-1.cdn.whatsapp.net
1810:127.0.0.1 media-bru2-1.cdn.whatsapp.net
1811:127.0.0.1 media-cdt1-1.cdn.whatsapp.net
1812:127.0.0.1 media-dfw5-1.cdn.whatsapp.net
1813:127.0.0.1 media-frt3-1.cdn.whatsapp.net
1814:127.0.0.1 media-frt3-2.cdn.whatsapp.net
1815:127.0.0.1 media-frx5-1.cdn.whatsapp.net
1816:127.0.0.1 media-iad3-1.cdn.whatsapp.net
1817:127.0.0.1 media-lax3-1.cdn.whatsapp.net
1818:127.0.0.1 media-lax3-2.cdn.whatsapp.net
1819:127.0.0.1 media-lga3-1.cdn.whatsapp.net
1820:127.0.0.1 media-lhr3-1.cdn.whatsapp.net
1821:127.0.0.1 media-lht6-1.cdn.whatsapp.net
1822:127.0.0.1 media-mad1-1.cdn.whatsapp.net
1823:127.0.0.1 media-mia3-1.cdn.whatsapp.net
1824:127.0.0.1 media-mxp1-1.cdn.whatsapp.net
1825:127.0.0.1 media-ort2-1.cdn.whatsapp.net
1826:127.0.0.1 media-ort2-2.cdn.whatsapp.net
1827:127.0.0.1 media-otp1-1.cdn.whatsapp.net
1828:127.0.0.1 media-sea1-1.cdn.whatsapp.net
1829:127.0.0.1 media-sin6-1.cdn.whatsapp.net
1830:127.0.0.1 media-sjc3-1.cdn.whatsapp.net
1831:127.0.0.1 media-sof1-1.cdn.whatsapp.net
1832:127.0.0.1 media-vie1-1.cdn.whatsapp.net
1833:127.0.0.1 media-waw1-1.cdn.whatsapp.net
1834:127.0.0.1 media-yyz1-1-cdn.whatsapp.net
1835:127.0.0.1 media.fabj2-1.fna.whatsapp.net
1836:127.0.0.1 media.fada1-2.fna.whatsapp.net
1837:127.0.0.1 media.fads1-1.fna.whatsapp.net
1838:127.0.0.1 media.fbed1-2.fna.whatsapp.net
1839:127.0.0.1 media.fbel3-1.fna.whatsapp.net
1840:127.0.0.1 media.fcgh5-1.fna.whatsapp.net
1841:127.0.0.1 media.fcgk9-1.fna.whatsapp.net
1842:127.0.0.1 media.fdel11-1.fna.whatsapp.net
1843:127.0.0.1 media.fdel3-1.fna.whatsapp.net
1844:127.0.0.1 media.feoh3-1.fna.whatsapp.net
1845:127.0.0.1 media.fevn1-1.fna.whatsapp.net
1846:127.0.0.1 media.ffjr1-1.fna.whatsapp.net
1847:127.0.0.1 media.ffjr1-3.fna.whatsapp.net
1848:127.0.0.1 media.fgye1-1.fna.whatsapp.net
1849:127.0.0.1 media.fhel2-1.fna.whatsapp.net
1850:127.0.0.1 media.fhel3-1.fna.whatsapp.net
1851:127.0.0.1 media.fisu5-1.fna.whatsapp.net
1852:127.0.0.1 media.flim6-1.fna.whatsapp.net
1853:127.0.0.1 media.flos2-1.fna.whatsapp.net
1854:127.0.0.1 media.fmnl10-1.fna.whatsapp.net
1855:127.0.0.1 media.fplu6-1.fna.whatsapp.net
1856:127.0.0.1 media.fprg2-1.fna.whatsapp.net
1857:127.0.0.1 media.frec2-1.fna.whatsapp.net
1858:127.0.0.1 media.frix1-1.fna.whatsapp.net
1859:127.0.0.1 media.frix4-1.fna.whatsapp.net
1860:127.0.0.1 media.fskg1-1.fna.whatsapp.net
1861:127.0.0.1 media.fssa13-1.fna.whatsapp.net
1862:127.0.0.1 media.fsst1-2.fna.whatsapp.net
1863:127.0.0.1 media.fsyq1-1.fna.whatsapp.net
1864:127.0.0.1 media.ftpe4-1.fna.whatsapp.net
1865:127.0.0.1 media.fykz1-1.fna.whatsapp.net
1866:127.0.0.1 media.fyxd2-1.fna.whatsapp.net
1867:127.0.0.1 media.fzty2-1.fna.whatsapp.net
1868:127.0.0.1 media.whatsapp.com
1869:127.0.0.1 mme.whatsapp.net
1870:127.0.0.1 mme6.whatsapp.net
1871:127.0.0.1 mmg-fna.whatsapp.net
1872:127.0.0.1 mmg.whatsapp.net
1873:127.0.0.1 mmi244.whatsapp.net
1874:127.0.0.1 mmi269.whatsapp.net
1875:127.0.0.1 mmi479.whatsapp.net
1876:127.0.0.1 mmi483.whatsapp.net
1877:127.0.0.1 mmi651.whatsapp.net
1878:127.0.0.1 mmi701.whatsapp.net
1879:127.0.0.1 mmi731.whatsapp.net
1880:127.0.0.1 mms.whatsapp.net
1881:127.0.0.1 pps.whatsapp.net
1882:127.0.0.1 ratings.whatsapp.com
1883:127.0.0.1 translate-dev.whatsapp.com
1884:127.0.0.1 translate.whatsapp.com
1885:127.0.0.1 up.web.whatsapp.com
1886:127.0.0.1 v.whatsapp.com
1887:127.0.0.1 v.whatsapp.net
1888:127.0.0.1 w1.web.whatsapp.com
1889:127.0.0.1 w2.web.whatsapp.com
1890:127.0.0.1 w3.web.whatsapp.com
1891:127.0.0.1 w4.web.whatsapp.com
1892:127.0.0.1 w5.web.whatsapp.com
1893:127.0.0.1 w6.web.whatsapp.com
1894:127.0.0.1 w7.web.whatsapp.com
1895:127.0.0.1 w8.web.whatsapp.com
1896:127.0.0.1 w9.web.whatsapp.com
1897:127.0.0.1 web.whatsapp.com
1898:127.0.0.1 whatsapp-cdn-shv-01-arn2.fbcdn.net
1899:127.0.0.1 whatsapp-cdn-shv-01-atl1.fbcdn.net
1900:127.0.0.1 whatsapp-cdn-shv-01-bom1.fbcdn.net
1901:127.0.0.1 whatsapp-cdn-shv-01-cdg2.fbcdn.net
1902:127.0.0.1 whatsapp-cdn-shv-01-dft4.fbcdn.net
1903:127.0.0.1 whatsapp-cdn-shv-01-dfw1.fbcdn.net
1904:127.0.0.1 whatsapp-cdn-shv-01-dfw5.fbcdn.net
1905:127.0.0.1 whatsapp-cdn-shv-01-frt3.fbcdn.net
1906:127.0.0.1 whatsapp-cdn-shv-01-gru2.fbcdn.net
1907:127.0.0.1 whatsapp-cdn-shv-01-hkg3.fbcdn.net
1908:127.0.0.1 whatsapp-cdn-shv-01-iad3.fbcdn.net
1909:127.0.0.1 whatsapp-cdn-shv-01-kut2.fbcdn.net
1910:127.0.0.1 whatsapp-cdn-shv-01-lax1.fbcdn.net
1911:127.0.0.1 whatsapp-cdn-shv-01-lax3.fbcdn.net
1912:127.0.0.1 whatsapp-cdn-shv-01-lga1.fbcdn.net
1913:127.0.0.1 whatsapp-cdn-shv-01-lhr3.fbcdn.net
1914:127.0.0.1 whatsapp-cdn-shv-01-mad1.fbcdn.net
1915:127.0.0.1 whatsapp-cdn-shv-01-mia1.fbcdn.net
1916:127.0.0.1 whatsapp-cdn-shv-01-ord1.fbcdn.net
1917:127.0.0.1 whatsapp-cdn-shv-01-sea1.fbcdn.net
1918:127.0.0.1 whatsapp-cdn-shv-01-sin1.fbcdn.net
1919:127.0.0.1 whatsapp-cdn-shv-01-sin6.fbcdn.net
1920:127.0.0.1 whatsapp-cdn-shv-01-sit4.fbcdn.net
1921:127.0.0.1 whatsapp-cdn-shv-01-sjc2.fbcdn.net
1922:127.0.0.1 whatsapp-cdn-shv-01-vie1.fbcdn.net
1923:127.0.0.1 whatsapp-cdn-shv-02-sin6.fbcdn.net
1924:127.0.0.1 whatsapp.com
1925:127.0.0.1 www-cdn.whatsapp.net
1926:127.0.0.1 www.whatsapp.com
1927:127.0.0.1 wwwl.whatsapp.com
```

**Comment by Lartsch** [comment]
There is also the possibility that is actually blocked by facebook hosts (and then further the possibility that Facebook has changed something in the WA client recently). As far as I can see on my end, WhatsApp client has only connected to several *.facebook.com hosts.

**Comment by schlaubstar** [comment]
I at least can confirm the fact, that using the social media extension still breaks the capability of WhatsApp to send/receive videos/pictures etc., today.

**Comment by dnmTX** [comment]
@schlaubstar  that is the whole purpose of the **social media extension**: TO BLOCK IT !!!!
https://github.com/StevenBlack/hosts#extensions

**Comment by schlaubstar** [comment]
@dnmTX I think you are wrong, it's a bug.
The ability of WhatsApp to send and receive media is an essential feature of the app. If you do not want to use it, don't use WhatsApp, but their cdn servers should not be containing the filter list because it is more than doubtful, that those servers are affecting you at any other place except for the app itself ... It makes a probably good  filter list completely useless for almost any android user

**Comment by Lartsch** [comment]
> @dnmTX I think you are wrong, it's a bug.
> The ability of WhatsApp to send and receive media is an essential feature of the app. If you do not want to use it, don't use WhatsApp, but their cdn servers should not be contained in the filter list because it is more than doubtful, that those servers are affecting you at any other place except for the app itself ... It makes a probably good filter list completely useless for the vast majority of android user

I support this view. I guess more than 4/5 of users of the social media extension still want to use Whatsapp in the end. But still, this is quite hard to make out since its not clear which servers are actually blocking it (facebook  and whatsapp servers involved)

**Comment by StevenBlack** [comment]
I'm tempted to close this thread because because **nobody** has told us, what specific domain needs to be unblocked so this specific feature of this specific app owned by Facebook can work.

Bringing this back to first principles: Facebook is an enormous, cancerous social media company.  Whatsapp is owned by Facebook.  Therefore, the social media extension should block **all** of Whatsapp.  That's my present inclination.

**Comment by funilrys** [comment]
Hi Steve @StevenBlack, I think this might help everybody who want to have it whitelisted (locally of course) :smile_cat: 

https://github.com/ukanth/afwall/wiki/HOWTO-blocking-WhatsApp

Cheers,
Nissar

**Comment by schlaubstar** [comment]
As I am reading the comments, I may explain a little more on my context. Your filterlist is included in the Blokada App, as a standard Provider for "FilterLists". The Blokada App itself sets up a VPN and filters/redirects DNS Queries based on those lists, thus having a system wide effect (which is a pretty nice approach if you do want to block adds systemwide on a non-rooted device).

If you just use it in a browser extension it would not bother me at all to have all Facebook/Whatsapp specific servers blocked. But: In combination with a systemwide DNS Filter, it is useless, as it breaks essential features of Whatsapp itself and in this case, local "adjustmens" are hard to realize as the process of importing the filterlist to blokada is highly automated and I would have to download the list, remove whatsapp servers, upload it again to an accessible place and then refer to this as a custom list for import which is quiet annoying.  

 I could not provide an IP, yet, because when trying it again it worked because the Server: qo-in-f188.1e100.net (173.194.66.188) is not part of the social-list

**Comment by glowinthedark** [comment]
Naively filtering the blocklist by the `*whatsapp*` pattern does NOT solve the media problem in mobile WhatsApp. There are some other hosts used for media and first time registration.

**Comment by spirillen** [comment]
My 2cents here is as @dnmTX I don't see the problem... the sole purpose of the "SOCIAL extension" is exactly to block all access to **_ANY_** online social site, so if you would like access to the Facebook/whatapp you have to unload the "Social list" from your hosts file integration, despite how you have included this.

**Comment by ak414** [comment]
done, deleting my hosts at ./etc/ but my device dont download / receive file media. 😣😣😣 i dont solved way to solving this. sadly

**Comment by Zyt3x** [comment]
Just why are you messing with host files still? It's just stenography.

Check the sourcecode for whatsapp and try sending a black.png pixel to whatever. It literary turns white on the travel and is stenographied back. end-to-end "encryption".

Just stop messing around with host files for social media. One is enough.

**Comment by abhinav96** [comment]
> I'm tempted to close this thread because **nobody** has told us, what specific domain needs to be unblocked so this specific feature of this specific app owned by Facebook can work.
> 
> Bringing this back to first principles: Facebook is an enormous, cancerous social media company.  Whatsapp is owned by Facebook.  Therefore, the social media extension should block **all** of Whatsapp.  That's my present inclination.
> 

Unblocking "mmg.whatsapp.net" fixed the media download issue for me. I didn't have issues with uploading media in the first place.

**Comment by Artek52** [comment]
I agree that blocking whatsapp doesn't making any sense as it is not a social media platform
